app/src/main/java/com/example/voyagerx/ui/fragments/userprofileac/FavoritesAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.voyagerx.ui.fragments.userprofileac
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.voyagerx.databinding.FavoritesOverviewCardBinding
+import com.example.voyagerx.repository.model.User
+import com.example.voyagerx.ui.fragments.userprofileac.FavoritesAdapter.*
+
+
+class FavoritesAdapter(
+    val user: User?) :
+    RecyclerView.Adapter<ViewHolder>() {
+
+
+    class ViewHolder(val binding: FavoritesOverviewCardBinding) : RecyclerView.ViewHolder(binding.root)
+
+    override fun getItemCount(): Int {
+        return user?.favoriteLaunches!!.size
+    }
```

Comment:
It's possible for favoriteLaunches to be null so I would probably do a null check here too. Maybe something to the effect of: `return user?.favoriteLaunches?.size ?: 0`

Reply:
Oh! That makes perfect sense. Thank you for pointing out the Elvis operator usage - I actually think I can implement this elsewhere, too. Thanks, @VivianLi98!!